### DISCUSSION TITLE
Welcome to generate-react-cli Discussions!

### DISCUSSION THREAD

**Discussion by arminbro** [arminbro/generate-react-cli#41] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Maruz** [comment]
Great work with #40! 💯

**Reply by arminbro** [reply]
Thanks Maruz!

**Comment by dbollinger** [comment]
Howdy! I'm fairly new to React, coming from 6+ years working in Ember and its CLI tools, so I was relieved to discover this project deliver some similar functionality.  Thanks for this!

I was pretty surprised to find that create-react-app doesn't offer anything like this on its own (and doesn't even really acknowledge the lack of feature), but found this article you wrote to be very helpful: https://dev.to/arminbro/generate-react-cli-1ooh -- it might be worth copying some of that (or linking it) into the README for this project.

Cheers!

**Reply by arminbro** [reply]
Hi Howie!

I'm glad to hear that GRC has been helpful for you while making the switch to React. 

About a year and a half ago, my buddy at work kept saying how helpful it would be if we had a CLI for generating react components on the fly and that he didn't find a good one in the market that met our needs. 

So I decided to build one myself. I didn't even think people would use it. I just wanted to create something for us so we can be more productive. I'm thrilled to see that it's also helping others out there!

Good call on adding the article! I'll make sure to link to it in the README.

Best,
Armin

**Comment by phuongdk** [comment]
Hi everyone, I'm currently using generate-react-cli library for my project and I have some issues with it. Any ideas to address these three problems as follow:

1. The lib always uppercase my letter even if I don't want it. For ex: If I type component name like "productinfo" it always end up with "Productinfo" which I don't want.

2. The lib automatically creates a folder and nests all the files within it without an option to choose whether to create a folder or not.

3. Is there a way to create a custom template file that apart from "component, test, story, style, lazy" for a specific folder that I define, not all of them (not like the "default" section tutorial on the guide). For ex: utils.template.js, helper.template.js, customhook.template.js,...

**Comment by vinilana** [comment]
Is there a contribution guide? I've just submitted a PR but I'm not sure if I'm following the requirements to contribute here.
